### DISCUSSION TITLE
Why does $in stop existing after the first line of a `def`?

### DISCUSSION THREAD

**Discussion by webbedspace** [nushell/nushell#6839] [Q&A] [answered]
This doesn't work:
```
# printp
# Print, then pipe input
def printp [] {
  print $in
  $in
}
```
This works:
```
# printp
# Print, then pipe input
def printp [] {
  let $a = $in
  print $a
  $a
}
```
How exactly does $in "drop out of scope" after the first line? Is there a reason for this?

**Comment by fdncred** [comment, marked as answer]
`$in` collects the input into a value during its use. https://www.nushell.sh/book/pipelines.html#basics you can search our docs for $in and find some hits but nushell isn't 100% documented since it changes on a daily basis still.

**Reply by webbedspace** [reply]
So, it's only piped into the first line of a `def`? I guess that makes sense… even though it kind of breaks my intuitions to have a `$` variable "disappear" after one line. That and it gets annoying to have to `let` it in order to use it in two lines (either that or use `do`)…

**Reply by fdncred** [reply]
I don't think it really has anything to do with the first line of a def. It's about the first use of anything in a def. So, if you're on the 10th line of a def and nothing has happened, then def is still viable. However, if you have any type of command, `$in` will be overwritten and no longer refers to what was passed in, it will refer to whatever the previous command/line/etc sent out.
